src/stores/right-panel/RightPanelStoreIPanelState.ts

```diff
@@ -109,8 +109,8 @@ export function convertCardToStore(panelState: IRightPanelCard): IRightPanelCard
     delete panelStateThisRoomStored.verificationRequestPromise;
     delete panelStateThisRoomStored.member;
 
-    const storedCard = { state: panelStateThisRoomStored as IRightPanelCardStored, phase: panelState.phase };
-    return storedCard as IRightPanelCardStored;
+    const storedCard = { state: panelStateThisRoomStored, phase: panelState.phase };
+    return storedCard;
 }
 
 function convertStoreToCard(panelStateStore: IRightPanelCardStored, room: Room): IRightPanelCard {
```

Comment:
The `as any` at the top of both of these conversion functions also seems surprising, can we remove those as well?

Reply:
I think we cant:
```
const panelStateThisRoom = { ...panelStateStore?.state } as any;
    if (!!panelStateThisRoom.threadHeadEventId) {
        panelStateThisRoom.threadHeadEvent = room.findEventById(panelStateThisRoom.threadHeadEventId);
    }
```
The panelStateStore.state does not have the property `threadHeadEvent` it only has `threadHeadEventId`. So the linter would be mad.

Reply:
I just tried without `as any` and with `as IRightPanelCardState` both cause the linter to be unhappy. Any other way to make it less surprising?

Reply:
Okay, I'll check this out locally and see if there's a better way.

Reply:
Perhaps it's better to explicitly copy the safe keys at first, and then you don't need the `delete` lines either...?

```
export function convertCardToStore(panelState: IRightPanelCard): IRightPanelCardStored {
    const panelStateThisRoomStored: IRightPanelCardStateStored = {
        groupId: panelState?.state?.groupId,
        groupRoomId: panelState?.state?.groupRoomId,
        widgetId: panelState?.state?.widgetId,
        spaceId: panelState?.state?.spaceId,
        isInitialEventHighlighted: panelState?.state?.isInitialEventHighlighted,
    };
    if (!!panelState?.state?.threadHeadEvent?.getId()) {
        panelStateThisRoomStored.threadHeadEventId = panelState.state.threadHeadEvent.getId();
    }
    if (!!panelState?.state?.memberInfoEvent?.getId()) {
        panelStateThisRoomStored.memberInfoEventId = panelState.state.memberInfoEvent.getId();
    }
    if (!!panelState?.state?.initialEvent?.getId()) {
        panelStateThisRoomStored.initialEventId = panelState.state.initialEvent.getId();
    }
    if (!!panelState?.state?.member?.userId) {
        panelStateThisRoomStored.memberId = panelState.state.member.userId;
    }

    const storedCard = { state: panelStateThisRoomStored, phase: panelState.phase };
    return storedCard;
}
```

Also at the moment, this guards against every possible level being undefined... but don't we know that at least `panelState` and `panelState.state` should exist, and those can use regular `.` instead of `?.`?